pandas/_libs/tslibs/period.pyx

```diff
@@ -2590,10 +2590,13 @@ class Period(_Period):
             dt = value
             if freq is None:
                 raise ValueError('Must supply freq for datetime value')
+            if isinstance(dt, Timestamp):
+                nanosecond = dt.nanosecond
         elif util.is_datetime64_object(value):
             dt = Timestamp(value)
             if freq is None:
                 raise ValueError('Must supply freq for datetime value')
+            nanosecond = dt.nanosecond
```

Comment:
Do we have a test with `np.datetime64` objects passed? (Do they even support nanosecond resolution?)

Reply:
yes and yes